feat: add license check

olexii4 opened the pull request:
### What does this PR do?
Add license check.

### What issues does this PR fix?
<!-- Please include any related issue from the Eclipse Che repository (or from another issue tracker). -->
fixes https://github.com/eclipse-che/che/issues/23363

### How to test this PR?

### Does this PR contain changes that override default upstream Code-OSS behavior?
- [ ] the PR contains changes in the [code](https://github.com/che-incubator/che-code/tree/main/code) folder (you can skip it if your changes are placed in a che extension )
- [ ] the corresponding items were added to the [CHANGELOG.md](https://github.com/che-incubator/che-code/blob/main/.rebase/CHANGELOG.md) file
- [ ] rules for automatic `git rebase` were added to the [.rebase](https://github.com/che-incubator/che-code/tree/main/.rebase) folder

olexii4 committed:
feat: add license check

Signed-off-by: Oleksii Orel <oorel@redhat.com>

RomanNikitenko commented on code/.deps/problems.md:
```diff
@@ -0,0 +1,33 @@
+5. `@vscode/test-web@0.0.62`
+6. `@vscode/vscode-perf@0.0.19`
+7. `innosetup@6.0.5`
+8. `postcss-nesting@12.0.2`
```
Could you clarify what are the next steps for the `Unresolved` dependencies?

olexii4 replied:
@RomanNikitenko The file `code/.deps/problems.md` just for information. It shows current state. To remove libraries from unresolved dependencies list, we can harvest them using this link [harvest](https://clearlydefined.io/harvest) or add libraries as exclusions in `code/.deps/EXCLUDED/prod.md` or `code/.deps/EXCLUDED/dev.md` files.

For example: https://github.com/eclipse-che/che-dashboard/blob/main/.deps/EXCLUDED/dev.md

RomanNikitenko replied:
There are 12 unresolved deps with `@vscode` prefix - could you clarify **_why_** they are unresolved?
did you try to apply `harvest` approach for them?

olexii4 replied:
I have fixed it

vitaliy-guliy commented:
Honestly, I do not have the whole picture of what this PR does.

Could we instead of adding all those files, add only one GitHub action, that will be triggered by pushing to the main branch (or pull request)? The action should not block anything and its successful execution may be represented as a badge in the README file.

olexii4 commented:
> Honestly, I do not have the whole picture of what this PR does.
> 
> Could we instead of adding all those files, add only one GitHub action, that will be triggered by pushing to the main branch (or pull request)? The action should not block anything and its successful execution may be represented as a badge in the README file.

@vitaliy-guliy We have some information here [che-incubator/dash-licenses](https://github.com/che-incubator/dash-licenses/tree/main?tab=readme-ov-file#usage):

...
**Update dependency info**
The following command generates dependencies info of a project and then checks all found dependencies. It returns a non-zero exit code if any of them are restricted to use.

```sh
docker run --rm -t \
       -v ${PWD}/:/workspace/project  \
       quay.io/che-incubator/dash-licenses:next
```
As a result, this command creates the next files:

 - `prod.md` with the list of production dependencies;
 - `dev.md` which contains only build and test dependencies;
 - `problems.md` will be created if some dependencies are not covered with CQ, unnecessary excludes present, etc.

**Check dependencies**
If you just need to verify that all dependencies satisfy IP requirements, use the --check flag, like the following

```sh
docker run --rm -t \
       -v ${PWD}/:/workspace/project  \
       quay.io/che-incubator/dash-licenses:next --check
```
So, this command doesn't create any new files in the project directory (except a temporary one) but checks if the dependencies info is up-to-date and then validates all found dependencies. It returns a non-zero exit code if any of the dependencies are restricted to use.
...

  Files `prod.md` and  `dev.md` include license information for libraries. for example:


  # Production dependencies

| Packages | License | Resolved CQs |
| --- | --- | --- |
| `@vscode/sqlite3@5.1.8-vscode` | BSD-3-Clause | [clearlydefined](https://clearlydefined.io/definitions/npm/npmjs/@vscode/sqlite3/5.1.8-vscode) |
  ...

You can click a link with CQ and the next page will be opened

![Знімок екрана 2025-03-12 о 15 20 20](https://github.com/user-attachments/assets/6c883d3a-6bd8-4a35-85a0-30cb41cc9dd6)

RomanNikitenko reviewed (commented):
@olexii4 
I propose to add section that describes license check functionality to the readme file
you could take this comment as a basis https://github.com/che-incubator/che-code/pull/527#issuecomment-2717913527

olexii4 commented:
> @olexii4 I propose to add section that describes license check functionality to the readme file you could take this comment as a basis [#527 (comment)](https://github.com/che-incubator/che-code/pull/527#issuecomment-2717913527)

@RomanNikitenko I have fixed it:

![Знімок екрана 2025-04-29 о 18 18 28](https://github.com/user-attachments/assets/1877b006-dbd9-46e9-82f9-edc7e405d89f)

RomanNikitenko commented on README.md:
````diff
@@ -86,6 +86,32 @@ For a main branch:
 $ git diff upstream-code/main main:code
 ```
 
+#### License Check for CHE extensions
````
It doesn't look like a separate section, it looks like a **sub**section of the `Updates and branches` section:

<img width="339" alt="image" src="https://github.com/user-attachments/assets/e854b5f7-684a-4194-b5d5-776fc017196b" />

olexii4 replied:
@RomanNikitenko Thanks. It was a copy/paste typo. I have fixed it.

RomanNikitenko commented on README.md:
````diff
@@ -86,6 +86,32 @@ For a main branch:
 $ git diff upstream-code/main main:code
 ```
 
+## License Check for CHE extensions
````
```suggestion
## License Check for Eclipse Che extensions
```

RomanNikitenko commented:
@olexii4 
ubi9-based build is failing, I created an issue for that problem https://github.com/eclipse-che/che/issues/23424
https://github.com/che-incubator/che-code/pull/541 should fix the problem, waiting for the jobs...

RomanNikitenko commented:
@olexii4 
I've merged my fix, please fetch new changes from the main branch to fix che-code build in your PR

vitaliy-guliy commented:
My first attempt of launching `npm run license:generate` has been failed.

I created a workspace from https://github.com/olexii4/che-code/tree/CHE-23363 and installed node dependencies. Then  I added `"comment-json": "4.2.5"` dependency to `che-api`, installed dependency again and launched `npm run license:generate` command for `che-api`

![Screenshot from 2025-05-02 13-56-17](https://github.com/user-attachments/assets/11f64847-4860-4d63-a245-c102b5577f09)

olexii4 commented:
> My first attempt of launching `npm run license:generate` has been failed.
> 
> I created a workspace from https://github.com/olexii4/che-code/tree/CHE-23363 and installed node dependencies. Then I added `"comment-json": "4.2.5"` dependency to `che-api`, installed dependency again and launched `npm run license:generate` command for `che-api`
> 
> ![Screenshot from 2025-05-02 13-56-17](https://private-user-images.githubusercontent.com/1655894/439891671-11f64847-4860-4d63-a245-c102b5577f09.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDYxOTA2OTQsIm5iZiI6MTc0NjE5MDM5NCwicGF0aCI6Ii8xNjU1ODk0LzQzOTg5MTY3MS0xMWY2NDg0Ny00ODYwLTRkNjMtYTI0NS1jMTAyYjU1NzdmMDkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDUwMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA1MDJUMTI1MzE0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NzNiNWQ5NGJiMzU2NDNjNWIxYjZjZDBhZWIxNTgzMDUzZjA5MTNlNDUxYTRkZjAyZTdlYzAwZGE2YzBkNmZhMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.lf6ObbI5wSWEA5iiEJQj47wL-a-9I1qzjgyjOR1m8IU)

@vitaliy-guliy I updated the thread. Please try again.

vitaliy-guliy commented:
> 
> @vitaliy-guliy I updated the thread. Please try again.

With the update podman doesn't even try to download the image

![Screenshot from 2025-05-02 18-14-51](https://github.com/user-attachments/assets/79bc9ef0-ebbc-46f7-b4e5-cb862cb0308b)

@olexii4 could you please update `package-lock.json` files for `che-*` extensions?
This is what I have after installing node dependencies
```
$ git status
On branch CHE-23363
Your branch is up to date with 'origin/CHE-23363'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   code/extensions/che-activity-tracker/package-lock.json
        modified:   code/extensions/che-api/package-lock.json
        modified:   code/extensions/che-commands/package-lock.json
        modified:   code/extensions/che-port/package-lock.json
        modified:   code/extensions/che-remote/package-lock.json
        modified:   code/extensions/che-resource-monitor/package-lock.json

no changes added to commit (use "git add" and/or "git commit -a")

```

olexii4 commented:
> > @vitaliy-guliy I updated the thread. Please try again.
> 
> With the update podman doesn't even try to download the image
> 
> ![Screenshot from 2025-05-02 18-14-51](https://private-user-images.githubusercontent.com/1655894/439965964-79bc9ef0-ebbc-46f7-b4e5-cb862cb0308b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDYxOTk2NjMsIm5iZiI6MTc0NjE5OTM2MywicGF0aCI6Ii8xNjU1ODk0LzQzOTk2NTk2NC03OWJjOWVmMC1lYmJjLTQ2ZjctYjRlNS1jYjg2MmNiMDMwOGIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDUwMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA1MDJUMTUyMjQzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NmNjYjllYjhjNDY5MjEwNjE0OTI2MzA3YTEzZDY2M2I3NzA2Y2ZmMTU0YTAyMjE0MWQ2YTBhY2RhNDY5NmIwYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.EouWdoTTJPBynQzRJ8Pfz7anO7W4S4Vad3Y_Q1VB9OU)
> 
> @olexii4 could you please update `package-lock.json` files for `che-*` extensions? This is what I have after installing node dependencies
> 
> ```
> $ git status
> On branch CHE-23363
> Your branch is up to date with 'origin/CHE-23363'.
> 
> Changes not staged for commit:
>   (use "git add <file>..." to update what will be committed)
>   (use "git restore <file>..." to discard changes in working directory)
>         modified:   code/extensions/che-activity-tracker/package-lock.json
>         modified:   code/extensions/che-api/package-lock.json
>         modified:   code/extensions/che-commands/package-lock.json
>         modified:   code/extensions/che-port/package-lock.json
>         modified:   code/extensions/che-remote/package-lock.json
>         modified:   code/extensions/che-resource-monitor/package-lock.json
> 
> no changes added to commit (use "git add" and/or "git commit -a")
> ```

@vitaliy-guliy Why should I do it in this PR?  It is a separate issue that doesn't depend on my changes.

vitaliy-guliy commented:
> > @olexii4 could you please update `package-lock.json` files for `che-*` extensions? This is what I have after installing node dependencies
> > ```
> > $ git status
> > On branch CHE-23363
> > Your branch is up to date with 'origin/CHE-23363'.
> > 
> > Changes not staged for commit:
> >   (use "git add <file>..." to update what will be committed)
> >   (use "git restore <file>..." to discard changes in working directory)
> >         modified:   code/extensions/che-activity-tracker/package-lock.json
> >         modified:   code/extensions/che-api/package-lock.json
> >         modified:   code/extensions/che-commands/package-lock.json
> >         modified:   code/extensions/che-port/package-lock.json
> >         modified:   code/extensions/che-remote/package-lock.json
> >         modified:   code/extensions/che-resource-monitor/package-lock.json
> > 
> > no changes added to commit (use "git add" and/or "git commit -a")
> > ```
> 
> @vitaliy-guliy Why should I do it in this PR? It is a separate issue that doesn't depend on my changes.

Ok, let's skip them as the changes are not related to this PR.

olexii4 commented:
<img width="1327" alt="Знімок екрана 2025-05-02 о 23 28 11" src="https://github.com/user-attachments/assets/6348b39f-c0b7-40d7-b9f7-596e455038b2" />

olexii4 commented:
@vitaliy-guliy FYI: https://github.com/che-incubator/dash-licenses/pull/26 was merged.

vitaliy-guliy commented:
Let's merge this PR as it is.
In case of any issues, changes in this PR will not block creating and merging the pull requests and will not affect the main branch.

olexii4 committed:
feat: add license check

Signed-off-by: Oleksii Orel <oorel@redhat.com>

vitaliy-guliy commented:
@olexii4 thanks for the contribution!

devspacesbuild commented:
Build [3.22](https://github.com/redhat-developer/devspaces-images/tree/devspaces-3-rhel-9/) :: [code_3.x/1597](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/code_3.x/1597/): [Console](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/code_3.x/1597/console), [Changes](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/code_3.x/1597/changes), [Git Data](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/code_3.x/1597/git)

devspacesbuild commented:
Build [3.22](https://github.com/redhat-developer/devspaces-images/tree/devspaces-3-rhel-9/) :: [sync-to-downstream_3.x/9332](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/sync-to-downstream_3.x/9332/): [Console](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/sync-to-downstream_3.x/9332/console), [Changes](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/sync-to-downstream_3.x/9332/changes), [Git Data](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/sync-to-downstream_3.x/9332/git)

devspacesbuild commented:
Build [3.22](https://github.com/redhat-developer/devspaces-images/tree/devspaces-3-rhel-9/) :: [get-sources-rhpkg-container-build_3.x/9480](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/get-sources-rhpkg-container-build_3.x/9480/): [Console](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/get-sources-rhpkg-container-build_3.x/9480/console), [Changes](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/get-sources-rhpkg-container-build_3.x/9480/changes), [Git Data](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/get-sources-rhpkg-container-build_3.x/9480/git)

devspacesbuild commented:
Build [3.22](https://github.com/redhat-developer/devspaces-images/tree/devspaces-3-rhel-9/) :: [push-latest-container-to-quay_3.x/5176](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/push-latest-container-to-quay_3.x/5176/): [Console](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/push-latest-container-to-quay_3.x/5176/console), [Changes](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/push-latest-container-to-quay_3.x/5176/changes), [Git Data](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/push-latest-container-to-quay_3.x/5176/git)

devspacesbuild commented:
Build [3.22](https://github.com/redhat-developer/devspaces-images/tree/devspaces-3-rhel-9/) :: [get-sources-rhpkg-container-build_3.x/9480](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/get-sources-rhpkg-container-build_3.x/9480/): [![SUCCESS](https://img.shields.io/badge/Build-SUCCESS-brightgreen?style=plastic&logo=redhat)](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/get-sources-rhpkg-container-build_3.x/9480/)<br /><blockquote>code : 3.x :: <a href='https://brewweb.engineering.redhat.com/brew/taskinfo?taskID=67663476'> Build  67663476</a> : <a href=https://quay.io/devspaces/code-rhel9:3.22-5>quay.io/devspaces/code-rhel9:3.22-5</a> <br /> SUCCESS; copied to quay; <a href=https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/push-latest-container-to-quay_3.x/5176/>/DS_CI/push-latest-container-to-quay_3.x/5176</a> triggered</blockquote>

devspacesbuild commented:
Build [3.22](https://github.com/redhat-developer/devspaces-images/tree/devspaces-3-rhel-9/) :: [update-digests_3.x/9233](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/update-digests_3.x/9233/): [Console](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/update-digests_3.x/9233/console), [Changes](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/update-digests_3.x/9233/changes), [Git Data](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/update-digests_3.x/9233/git)

devspacesbuild commented:
Build [3.22](https://github.com/redhat-developer/devspaces-images/tree/devspaces-3-rhel-9/) :: [code_3.x/1597](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/code_3.x/1597/): [![SUCCESS](https://img.shields.io/badge/Build-SUCCESS-brightgreen?style=plastic&logo=redhat)](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/code_3.x/1597/)<br /><blockquote>Upstream sync done; <a href=https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/sync-to-downstream_3.x/9332/>/DS_CI/sync-to-downstream_3.x/9332</a> triggered</blockquote>

devspacesbuild commented:
Build [3.22](https://github.com/redhat-developer/devspaces-images/tree/devspaces-3-rhel-9/) :: [operator-bundle_3.x/4909](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/operator-bundle_3.x/4909/): [Console](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/operator-bundle_3.x/4909/console), [Changes](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/operator-bundle_3.x/4909/changes), [Git Data](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/operator-bundle_3.x/4909/git)

devspacesbuild commented:
Build [3.22](https://github.com/redhat-developer/devspaces-images/tree/devspaces-3-rhel-9/) :: [sync-to-downstream_3.x/9333](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/sync-to-downstream_3.x/9333/): [Console](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/sync-to-downstream_3.x/9333/console), [Changes](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/sync-to-downstream_3.x/9333/changes), [Git Data](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/sync-to-downstream_3.x/9333/git)

devspacesbuild commented:
Build [3.22](https://github.com/redhat-developer/devspaces-images/tree/devspaces-3-rhel-9/) :: [get-sources-rhpkg-container-build_3.x/9481](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/get-sources-rhpkg-container-build_3.x/9481/): [Console](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/get-sources-rhpkg-container-build_3.x/9481/console), [Changes](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/get-sources-rhpkg-container-build_3.x/9481/changes), [Git Data](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/get-sources-rhpkg-container-build_3.x/9481/git)

devspacesbuild commented:
Build [3.22](https://github.com/redhat-developer/devspaces-images/tree/devspaces-3-rhel-9/) :: [push-latest-container-to-quay_3.x/5177](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/push-latest-container-to-quay_3.x/5177/): [Console](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/push-latest-container-to-quay_3.x/5177/console), [Changes](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/push-latest-container-to-quay_3.x/5177/changes), [Git Data](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/push-latest-container-to-quay_3.x/5177/git)

devspacesbuild commented:
Build [3.22](https://github.com/redhat-developer/devspaces-images/tree/devspaces-3-rhel-9/) :: [copyIIBsToQuay/2999](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/Releng/job/copyIIBsToQuay/2999/): [Console](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/Releng/job/copyIIBsToQuay/2999/console), [Changes](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/Releng/job/copyIIBsToQuay/2999/changes), [Git Data](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/Releng/job/copyIIBsToQuay/2999/git)

devspacesbuild commented:
Build [3.22](https://github.com/redhat-developer/devspaces-images/tree/devspaces-3-rhel-9/) :: [get-sources-rhpkg-container-build_3.x/9481](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/get-sources-rhpkg-container-build_3.x/9481/): [![SUCCESS](https://img.shields.io/badge/Build-SUCCESS-brightgreen?style=plastic&logo=redhat)](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/get-sources-rhpkg-container-build_3.x/9481/)<br /><blockquote>devspaces-operator-bundle : 3.x :: <a href='https://brewweb.engineering.redhat.com/brew/taskinfo?taskID=67665664'> Build  67665664</a> : <a href=https://quay.io/devspaces/devspaces-operator-bundle:3.22-13>quay.io/devspaces/devspaces-operator-bundle:3.22-13</a> <br /> SUCCESS; copied to quay; <a href=https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/push-latest-container-to-quay_3.x/5177/>/DS_CI/push-latest-container-to-quay_3.x/5177</a> triggered;<br /><blockquote>Bundle <a href=https://github.com/redhat-developer/devspaces-images/blob/devspaces-3-rhel-9/devspaces-operator-bundle-generated/manifests/devspaces.csv.yaml>CSV</a> related images:<br />- <a href='https://quay.io/devspaces-tech-preview/idea-rhel9:'>quay.io/devspaces-tech-preview/idea-rhel9:</a><br />- <a href='https://quay.io/devspaces-tech-preview/jetbrains-ide-rhel9:'>quay.io/devspaces-tech-preview/jetbrains-ide-rhel9:</a><br />- <a href='https://quay.io/devspaces/code-rhel9:3.22-5'>quay.io/devspaces/code-rhel9:3.22-5</a><br />- <a href='https://quay.io/devspaces/configbump-rhel9:3.22-2'>quay.io/devspaces/configbump-rhel9:3.22-2</a><br />- <a href='https://quay.io/devspaces/dashboard-rhel9:3.22-7'>quay.io/devspaces/dashboard-rhel9:3.22-7</a><br />- <a href='https://quay.io/devspaces/devspaces-rhel9-operator:3.22-2'>quay.io/devspaces/devspaces-rhel9-operator:3.22-2</a><br />- <a href='https://quay.io/devspaces/pluginregistry-rhel9:3.22-2'>quay.io/devspaces/pluginregistry-rhel9:3.22-2</a><br />- <a href='https://quay.io/devspaces/server-rhel9:3.22-5'>quay.io/devspaces/server-rhel9:3.22-5</a><br />- <a href='https://quay.io/devspaces/traefik-rhel9:3.22-1'>quay.io/devspaces/traefik-rhel9:3.22-1</a><br />- <a href='https://quay.io/devspaces/udi-rhel9:3.22-3'>quay.io/devspaces/udi-rhel9:3.22-3</a><br />= <a href='https://catalog.redhat.com/software/containers/search?q=openshift4/ose-kube-rbac-proxy'>registry.redhat.io/openshift4/ose-kube-rbac-proxy:v4.14.0-202504220036.p0.gb8b8259.assembly.stream.el8</a><br />= <a href='https://catalog.redhat.com/software/containers/search?q=openshift4/ose-oauth-proxy'>registry.redhat.io/openshift4/ose-oauth-proxy:v4.14.0-202504220036.p0.ga4a2f27.assembly.stream.el8</a><br />= <a href='https://catalog.redhat.com/software/containers/search?q=ubi8/ubi-minimal'>registry.redhat.io/ubi8/ubi-minimal:8.8-1072.1697626218</a></blockquote></blockquote>

devspacesbuild commented:
Build [3.22](https://github.com/redhat-developer/devspaces-images/tree/devspaces-3-rhel-9/) :: [sync-to-downstream_3.x/9333](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/sync-to-downstream_3.x/9333/): [![SUCCESS](https://img.shields.io/badge/Build-SUCCESS-brightgreen?style=plastic&logo=redhat)](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/sync-to-downstream_3.x/9333/)<br /><blockquote>Build container: devspaces-operator-bundle synced;  <a href=https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/get-sources-rhpkg-container-build_3.x/9481/>/DS_CI/get-sources-rhpkg-container-build_3.x/9481</a> triggered;  <a href=https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/dsc_3.x/>/job/DS_CI/job/dsc_3.x</a> triggered; </blockquote>

devspacesbuild commented:
Build [3.22](https://github.com/redhat-developer/devspaces-images/tree/devspaces-3-rhel-9/) :: [operator-bundle_3.x/4909](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/operator-bundle_3.x/4909/): [![SUCCESS](https://img.shields.io/badge/Build-SUCCESS-brightgreen?style=plastic&logo=redhat)](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/operator-bundle_3.x/4909/)<br /><blockquote>Upstream sync done; <a href=https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/sync-to-downstream_3.x/9333/>/DS_CI/sync-to-downstream_3.x/9333</a> triggered</blockquote>

devspacesbuild commented:
Build [3.22](https://github.com/redhat-developer/devspaces-images/tree/devspaces-3-rhel-9/) :: [dsc_3.x/2054](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/dsc_3.x/2054/): [Console](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/dsc_3.x/2054/console), [Changes](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/dsc_3.x/2054/changes), [Git Data](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/dsc_3.x/2054/git)

devspacesbuild commented:
Build [3.22](https://github.com/redhat-developer/devspaces-images/tree/devspaces-3-rhel-9/) :: [update-digests_3.x/9233](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/update-digests_3.x/9233/): [![SUCCESS](https://img.shields.io/badge/Build-SUCCESS-brightgreen?style=plastic&logo=redhat)](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/update-digests_3.x/9233/)<br /><blockquote>Detected new images: rebuild operator-bundle<br />* code; <a href=https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/operator-bundle_3.x/4909/>/DS_CI/operator-bundle_3.x/4909</a> triggered</blockquote>

devspacesbuild commented:
Build [3.22](https://github.com/redhat-developer/devspaces-images/tree/devspaces-3-rhel-9/) :: [dsc_3.x/2054](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/dsc_3.x/2054/): [![SUCCESS](https://img.shields.io/badge/Build-SUCCESS-brightgreen?style=plastic&logo=redhat)](https://jenkins-csb-crwqe-main.dno.corp.redhat.com/job/DS_CI/job/dsc_3.x/2054/)<br /><blockquote><a href=https://quay.io/devspaces/dsc:3.22.0-CI>3.22.0-CI</a></blockquote>
